packages/@aws-cdk/region-info/lib/region-info-token.ts

```diff
@@ -0,0 +1,30 @@
+import cdk = require('@aws-cdk/cdk');
+import { RegionInfo } from './region-info';
+
+/**
+ * A Token that rocudes a value from looking up a RegionInfo fact based on the resolution context's stack region.
+ */
+export class RegionInfoToken extends cdk.Token {
```

Comment:
This makes this library depend on `@aws-cdk/cdk`. Would have preferred if the dependency would be the other way around: this library is just a facts library without dependencies, and CDK core depends on it.

That's also the only way to structure the dependency graph if we ever want to make decisions during stack synthesis based on the info in here.

Also, toolkit will depend on this library (for metadata service presence) and then transitively it would also depend on `@aws-cdk/cdk`, even though it really shouln't have a reason to?

Reply:
I guess I can move the Token to the `@aws-cdk/cdk` library... But I'd leave the "built-in" facts where they are now... Sounds good?

Reply:
Agree with @rix0rrr that we don't want this dependency. I'd argue that this library should not be part of the CDK at all. Why not just ask the user to pass in `region` when they interact with this library that's centered around region information?